generic_config_updater/patch_sorter.py

```diff
@@ -290,29 +290,52 @@ def __hash__(self):
         return hash((self.op_type, self.path, json.dumps(self.value)))
 
 class MoveWrapper:
-    def __init__(self, move_generators, move_extenders, move_validators):
+    def __init__(self, move_generators, move_non_extendable_generators, move_extenders, move_validators):
         self.move_generators = move_generators
+        self.move_non_extendable_generators = move_non_extendable_generators
         self.move_extenders = move_extenders
         self.move_validators = move_validators
 
     def generate(self, diff):
+        """
+        Generates all possible moves to help transform diff.current_config to diff.target_config.
+
+        It starts by generating the non-extendable moves i.e. moves that will not extended to e.g. change its parent.
+        The non-extendable moves are mostly high level moves such as deleting/adding whole tables.
+
+        After that it generates extendable moves i.e. moves that can be extended to e.g. change its parent.
+        The extendable moves are typically very low level moves that can achieve the minimum disruption guarantee.
+
+        Lastly the moves are extended for example to try to replace the parent config instead, or by deleting
+        the dependencies of the config.
+        """
         processed_moves = set()
+        extended_moves = set()
         moves = deque([])
 
+        for move in self._generate_non_extendable_moves(diff):
+            if not(move in processed_moves):
```

Comment:
>not(move in processed_moves [](http://example.com/codeflow?start=15&length=27)

Minor suggestion:
```
move not in processed_moves
```

Reply:
I have another PR to to hide log printing from yanglib, will include this fix as well.